tf-static-website/23-cloudfront-frontend.tf

```diff
@@ -0,0 +1,65 @@
+resource "aws_cloudfront_origin_access_identity" "origin_access_identity" {
+  comment  = "This is used for my website."
+  provider = aws.ops-blog
+}
+
+locals {
+  s3_origin_id = "myS3Origin"
+}
+
+resource "aws_cloudfront_distribution" "s3_distribution" {
+  origin {
+    domain_name = aws_s3_bucket.ops_website.bucket_regional_domain_name
+    origin_id   = local.s3_origin_id
+
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.origin_access_identity.cloudfront_access_identity_path
+    }
+  }
+
+  enabled             = true
+  comment             = "This is used for my website."
+  default_root_object = "index.html"
+
+  logging_config {
+    include_cookies = false
+    bucket          = aws_s3_bucket.ops_website.bucket_regional_domain_name
+    prefix          = "access_log"
+  }
+
+  aliases = [ var.domain_name ]
+
+  default_cache_behavior {
+    allowed_methods  = ["GET", "HEAD", "OPTIONS"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = local.s3_origin_id
+
+    forwarded_values {
+      query_string = false
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "redirect-to-https"
+    min_ttl                = 0
+    default_ttl            = 3600
+    max_ttl                = 86400
+  }
+
+  restrictions {
+    geo_restriction {
+      restriction_type = "none"
+    }
+  }
+
+  viewer_certificate {
+    ssl_support_method       = "sni-only"
+    acm_certificate_arn      = aws_acm_certificate.cert.arn
+    minimum_protocol_version = "TLSv1"
```

Comment:
As of now, TLSv1 is vulnerable to man-in-the-middle attacks, which is not secure to users. I recommend you to specify here either: `TLSv1_2016`, `TLSv1.1_2016` or `TLSv1.2_2018`

Ref: https://www.terraform.io/docs/providers/aws/r/cloudfront_distribution.html#minimum_protocol_version

Reply:
I'll update it to use `TLSv1.1_2016`